documentation/projects/proposals/search_relevancy_sandbox/20230518-implementation_plan_staging_index_rapid_iteration.md

```diff
@@ -0,0 +1,380 @@
+# 2023-05-18 Implementation Plan: Rapid iteration of ingestion server index configuration
+
+**Author**: @aetherunbound
+
+<!-- See the implementation plan guide for more information: https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc -->
+<!-- This template is exhaustive and may include sections which aren't relevant to your project. Feel free to remove any sections which would not be useful to have. -->
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @sarayourfriend
+- [ ] @stacimc
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/392)
+- [Project Proposal](https://github.com/WordPress/openverse/issues/392)
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+This document describes a DAG (and associated dependencies) which can be used to
+generate a new index in Elasticsearch without requiring a data refresh,
+ingestion server deployment, or any interaction with the ingestion server.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+Once the implementation is complete, a maintainer should be able to run this DAG
+to create a new index on either elasticsearch environment with an altered
+configuration. The maintainer should only need to specify the pieces of the
+configuration that are different from the default configuration, although they
+can supply a full configuration if they wish. The DAG will report when it is
+complete and what the name of the new index is.
+
+## Dependencies
+
+The following are prerequisites for the DAG:
+
+- Installation of the
+  [`elasticsearch` Airflow provider package](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/)
+- Airflow Connections for each Elasticsearch cluster
+
+### Elasticsearch provider
+
+The
+[`elasticsearch` provider](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/)
+will need to be added to the list of dependencies for our catalog Docker image.
+This will be done first and deployed on its own, as the connections will be
+needed for the rest of the work.
+
+The
+[production requirements](https://github.com/WordPress/openverse/blob/3fcce5ade2165955db5bbcb4f679257b3260547b/catalog/requirements_prod.txt#L5)
+will need to be changed from `apache-airflow[amazon,postgres,http]` to
+`apache-airflow[amazon,postgres,http,elasticsearch]`.
```

Comment:
Are these two discrete steps? Can they be done at the same time? Or is the second line the directions for implementing the first bit? Sorry if this is obvious, the wording just leaves it a bit unclear :sweat_smile:

Reply:
The second paragraph contains the implementation directions for the first - I'll reword this to make that a bit more clear!